pkg/services/alerting/engine.go

```diff
@@ -99,22 +104,35 @@ func (e *Engine) processJob(grafanaCtx context.Context, job *Job) error {
 	}()
 
 	alertCtx, cancelFn := context.WithTimeout(context.Background(), alertTimeout)
+	span := opentracing.StartSpan("alert execution")
+	alertCtx = opentracing.ContextWithSpan(alertCtx, span)
 
 	job.Running = true
 	evalContext := NewEvalContext(alertCtx, job.Rule)
+	evalContext.Ctx = alertCtx
 
 	done := make(chan struct{})
-
 	go func() {
 		defer func() {
 			if err := recover(); err != nil {
 				e.log.Error("Alert Panic", "error", err, "stack", log.Stack(1))
+				ext.Error.Set(span, true)
+				span.LogFields(
+					tlog.Error(fmt.Errorf("%v", err)),
```

Comment:
why not simply `tlog.Error(err)` ?

Reply:
`tlog.Error` expects a parameter with type `error`. `recover()` returns an `interface{}`
So my choices are
* type assertion (I think that is to verbose)
* using `fmt.Errorf` that is using reflection behind the scenes.
* just cast it to error/string

I'm not sure what would be the best option. Any thoughts? 

ping @Dieterbe

Reply:
looks fine to me